ext/ldap/ldap.c

```diff
@@ -1016,6 +1016,77 @@ PHP_FUNCTION(ldap_connect)
 }
 /* }}} */
 
+#ifdef HAVE_ORALDAP
+#ifdef LDAP_API_FEATURE_X_OPENLDAP
```

Comment:
```suggestion
#if defined(HAVE_ORALDAP) && defined(LDAP_API_FEATURE_X_OPENLDAP)
```

Reply:
Makes sense. P.S. this is the only function where I didn't extract the common pieces of code with the original function... mostly because it's somewhat tedious due to `ifdefs` and the removed `port` parameter.